lib/msal-node/docs/caching.md

```diff
@@ -65,59 +67,61 @@ interface ICachePlugin {
-* For web APIs calling other web APIs using OBO: hash of the incoming access token from the user (i.e. the token which will subsequently be exchanged for an OBO token)
-
-MSAL Node provides the [DistributedCachePlugin](https://azuread.github.io/microsoft-authentication-library-for-js/ref/classes/_azure_msal_node.distributedcacheplugin.html) class for confidential clients, which implements the [ICachePlugin](https://azuread.github.io/microsoft-authentication-library-for-js/ref/interfaces/_azure_msal_common.icacheplugin.html). An instance of `DistributedCachePlugin` requires:
-
-* a **client interface**, which implements `get` and `set` operations on the persistence server (Redis, MySQL etc.).
-* a **partition manager**, for reading from and writing to cache with respect to a given **partition key**.
-
-Please refer to the [RedisTestApp](../../../samples/msal-node-samples/RedisTestApp/README.md) for a sample implementation.
-
-> :bulb: We recommend you to obtain performance metrics for cache operations, such as **cache retrieval duration** and **cache hit ratios** in your applications. This would give you a reliable picture of how your app scales over time. For instance, to measure how much time MSAL spends in `acquireToken*` calls, you can use Node's [performance measurement API](https://nodejs.org/api/perf_hooks.html)s along with the `fromCache` property in [AuthenticationResult](https://azuread.github.io/microsoft-authentication-library-for-js/ref/modules/_azure_msal_common.html#authenticationresult) class:
->
-> ```javascript
->    const { PerformanceObserver, performance } = require('node:perf_hooks');
->
->    const perfObserver = new PerformanceObserver((items) => {
->        items.getEntries().forEach((entry) => {
->            console.log(entry);
->        })
->    });
->
->    perfObserver.observe({ entryTypes: ["measure"], buffer: true });
->
->    // ...
->
->    performance.mark("acquireTokenSilent-start");
->    const tokenResponse = await msalInstance.acquireTokenSilent({
->        account: account,
->        scopes: ["User.Read"],
->    });
->    performance.mark("acquireTokenSilent-end");
->
->    const measurementName = tokenResponse.fromCache ? "acquireTokenSilent-fromCache" : "acquireTokenSilent-fromNetwork";
->    performance.measure(measurementName, "acquireTokenSilent-start", "acquireTokenSilent-end");
-> ```
->
-> You should collect these along with the performance metrics generated by your storage solution (e.g. Redis) used for persistence.
+On confidential client apps that handle users (web apps that sign in users and call web APIs, and web APIs calling downstream web APIs), there can be many users and the users are processed in parallel. Our recommendation is to serialize one cache blob (see [CacheRecord](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/src/cache/entities/CacheRecord.ts)) per user. Use a key for partitioning the cache (*i.e.* **partition key**), such as:
```

Comment:
```suggestion
On confidential client apps that handle users (web apps that sign in users and call web APIs, and web APIs calling downstream web APIs), there can be many users active concurrently for a given application. Our recommendation is to serialize one cache blob (see [CacheRecord](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-common/src/cache/entities/CacheRecord.ts)) per user. This would help with scaling the cache across a distributed system. Use a key for partitioning the cache (*i.e.* **partition key**), such as:
```